qiskit/primitives/containers/observables_array.py

```diff
@@ -211,7 +211,7 @@ def coerce(cls, observables: ObservablesArrayLike) -> ObservablesArray:
     def validate(self):
         """Validate the consistency in observables array."""
         num_qubits = None
-        for obs in self._array:
+        for obs in self._array.reshape(-1):
```

Comment:
I'm not sure why this change is included. I guess some validations may fail due to the previous version. It might be good to add a unit test of observable array.

Reply:
Yes, it was given an error about not being iterable when `shape=()`, and it was also not working properly when ndim>1. This reshape trick gurantees it to always be a vector. But you're right that I should have put a test in.

Reply:
A test was added in 12c71e5